gateleen-test/src/test/java/org/swisspush/gateleen/hook/RouteListingTest.java

```diff
@@ -300,15 +306,20 @@ public void testRouteListing_NoRoutesRegistered(TestContext context) {
         // No routes registered
 
         // Send GET request with a query param
-        given().queryParam("q", queryParam)
+        Response response = given().queryParam("q", queryParam)
                 .when().get(requestUrl)
                 .then().assertThat().statusCode(200)
-                .body("routes", Matchers.empty());
+                .extract().response();
+
+        // Print the body of the response for debugging
+        System.out.println("Response body: " + response.getBody().asString());
 
-        // Validate response
-        checkGETStatusCodeWithAwait(requestUrl, 200);
+        // Assert that the response body is empty or does not contain routes
+        Assert.assertTrue("No routes should be registered",
```

Comment:
What do you assert here? Your test will be green for outputs like

```json
{"routes":[]}
```
but also for
```json
{"routes":["foo","bar"]}
```
and that is probably not what you want to assert here

Reply:
Created a new validate to check it

Reply:
Where is the new validation to check the repsonse? This is still not fixed since my statements about the possible responses that satisfy your validation is still true.

Reply:
Add validation to check if contains the value and also check if the result is empty